### ISSUE TITLE
Setting the password doesn't appear to work

### ISSUE THREAD

**Issue by lukesmurray** [neo4j/docker-neo4j#147]
I've been trying to set an initial password on my Neo4j docker container but it doesn't appear to be doing anything. When I access the container the password is still set to the default which is neo4j. I'm passing in an environment variable based on the documentation I found (here)[https://neo4j.com/docs/operations-manual/current/installation/docker/]. When I try to set the password to something custom the password remains the default neo4j. However, if I say `NEO4J_AUTH=none` then the container respects that and uses no password.

This is the contents of my docker-compose.yml

```
version: "3.7"
services:
  neo4j:
    image: neo4j:latest
    container_name: neo4j
    volumes:
      - ./persisted_data/neo4j/data:/data
    ports:
      # http and bolt access respectively
      - 7474:7474
      - 7687:7687
    env_file:
      - .env
```

This is my .env file which works

```
# neo4j auth as username/password
NEO4J_AUTH=none
```

This is my .env file which fails

```
# neo4j auth as username/password
NEO4J_AUTH=neo4j/anything_here_is_replaced_by_neo4j_as_the_password
```

**Comment by jennyowen** [comment]
I found a setting password bug while fixing an unrelated issue, and I think it may have been causing this problem. 

You would hit the bug if you start the docker container without the `--user` flag and while setting a password, eg:
```
docker run \
    --publish=7474:7474 --publish=7687:7687 \
    --volume=$HOME/neo4j/data:/data \
    --volume=$HOME/neo4j/logs:/logs \
    --env=NEO4J_AUTH=neo4j/newpassword \
    neo4j:3.5
```

Running Neo4j with the `--user` flag does NOT hit the bug. eg:
```
docker run \
    --publish=7474:7474 --publish=7687:7687 \
    --volume=$HOME/neo4j/data:/data \
    --volume=$HOME/neo4j/logs:/logs \
    --env=NEO4J_AUTH=neo4j/newpassword \
    --user="$(id -u):$(id -g)" \
    neo4j:3.5
```
(See https://neo4j.com/docs/operations-manual/current/docker/introduction/#docker-user)

I fixed the bug in pull request #167 so it should be fixed when we release 3.5.6 for docker.
Until then, try adding the `--user` flag as a workaround.

I'm closing this as a duplicate of #163  so if you still have the same problem with the workaround or patched release, please post on that issue.

Also @lukesmurray  thanks for taking the time to report the problem, and I'm sorry we didn't fix it sooner.

**Comment by SaltedEggIndomee** [comment]
@jennyowen How about Kubernetes?? 

Where should be provide the workaround to include the --user flag??
